libraries/nexus_helper.rb

```diff
@@ -0,0 +1,42 @@
+require 'timeout'
+require 'open-uri'
+
+module Nexus3
+  # Helper library for testing Nexus3 responses
+  module Helper
+    # Raise an error in case Nexus takes a really long time to start.
+    class Nexus3NotReady < StandardError
+      def initialize(endpoint, timeout)
+        super <<-EOH
+The Nexus server at #{endpoint} did not become ready within #{timeout}
+seconds. On large Nexus instances (or on smaller hardware), you may need
+to increase the timeout to #{timeout * 4} seconds. Alternatively, Nexus
+may have failed to start.
+EOH
+      end
+    end
+
+    def wait_until_ready!(endpoint)
```

Comment:
Done, with a default and called with node['nexus3']['api']['wait'] (the 'wait' attribute name comes from the original cookbook and I think we should not necessarily rename it right away.)